File: src/main/java/io/dddrive/doc/model/DocPartTransition.java
package io.dddrive.doc.model;

import java.time.OffsetDateTime;

import io.dddrive.doc.model.enums.CodeCaseStage;
import io.dddrive.oe.model.ObjUser;

public interface DocPartTransition extends DocPart<Doc> {

	Integer getSeqNr();

	ObjUser getUser();

	OffsetDateTime getTimestamp();

	CodeCaseStage getOldCaseStage();

	CodeCaseStage getNewCaseStage();

}

File: fm-dddrive/src/main/java/dddrive/app/doc/model/DocPartTransition.kt
package dddrive.app.doc.model

import dddrive.app.doc.model.enums.CodeCaseStage
import java.time.OffsetDateTime

interface DocPartTransition : DocPart<Doc> {

	val seqNr: Int

	val userId: Any

	// val user: ObjUser

	val timestamp: OffsetDateTime

	val oldCaseStage: CodeCaseStage?

	val newCaseStage: CodeCaseStage

	fun init(
		userId: Any,
		timestamp: OffsetDateTime,
		oldCaseStage: CodeCaseStage?,
		caseStage: CodeCaseStage,
	)

}
